### PR TITLE
Differentiate between production and dev environments

### PR DESCRIPTION
**Pull request by p-galligan**
Adds a second config file for development purposes, and tweaks the build script.

So, working off the assumption that we can set `jekyll_env` on the server side, all we'd need to do is set it to `production` on the production server and it'll run off of `_config.yml` settings.

Otherwise, any other environment setting will use the `_config_dev.yml` settings when building. The settings loaded from the later config will override settings in the earlier one.

LMK if you think this won't work for our deployment.

### CODE REVIEW

File: script/build [hunk 1]
```diff
@@ -3,7 +3,13 @@ set -e # halt script on error
 
 python script/make_pages.py
 
-bundle exec jekyll build
+if JEKYLL_ENV=production; then
```
Comment (helrond) [inline script/build]:
```suggestion
if [ "$JEKYLL_ENV" = "production"]; then
```

File: script/build [hunk 2]
```diff
@@ -3,7 +3,13 @@ set -e # halt script on error
...
+  bundle exec jekyll build --config _config.yml
+  echo "production"
+else
+  bundle exec jekyll build --config _config.yml,_config_dev.yml
```
Comment (helrond) [inline script/build]:
```suggestion
  bundle exec jekyll build --config _config_dev.yml
```

**Review by helrond — CHANGES_REQUESTED** [review]

File: script/build [hunk 3]
```diff
@@ -3,7 +3,13 @@ set -e # halt script on error
 
 python script/make_pages.py
 
-bundle exec jekyll build
+if [ "$JEKYLL_ENV" = "production"]; then
```
Comment (helrond) [inline script/build]:
```suggestion
if [ "$JEKYLL_ENV" = "production" ]; then
```

**Comment by helrond** [comment]
I think this will work. I'm wondering, though, about how wise it is to have two config files with almost identical values. If you pass multiple configs, do those values get merged? Would it be possible to have a _config.yml that has all the base values and then a _config.prod.yml that has ONLY the site.url value?

**Comment by p-galligan** [comment]
@helrond As far as I can tell. I think your suggestion would work. "Settings in later files override settings in earlier files." So I _think_ that it should work to overwrite, and duplication isn't necessary either one way or another.

**Comment by helrond** [comment]
OK, so let's try that then?

**Comment by p-galligan** [comment]
Unfortunately it doesn't selectively overwrite... I just tested locally and it needed a value for every variable in both configs.

**Review by helrond — APPROVED** [review]

**Comment by helrond** [comment]
Maybe just get rid of the `echo` lines in script/build?

**Comment by p-galligan** [comment]
@helrond Yep, I have that committed locally, I just haven't pushed yet. Do you mind if I add the comment to `make_pages` and the readme change here as well?

**Comment by helrond** [comment]
That makes sense - let's get that all in this PR.

**Pull request merged**